src/rust/engine/process_execution/src/local.rs

```diff
@@ -76,10 +77,36 @@ impl super::CommandRunner for CommandRunner {
               })
                   .map(|posix_fs| Arc::new(posix_fs))
                   .and_then(|posix_fs| {
```

Comment:
The nesting has gotten pretty deep here... would suggest breaking the body of this `and_then` into a new function/method.

Reply:
done - let me know if you think there might've been a better way to do this.